src/js/lib/content/src/index.js

```diff
@@ -122,6 +122,101 @@ function show_debug_screen(wasm,msg) {
 
 
 
+// ======================
+// === Crash Handling ===
+// ======================
+
+function recoveringFromCrash() {
+    return sessionStorage.getItem("crash-message") !== null
+}
+
+function previousCrashMessage() {
+    return sessionStorage.getItem("crash-message")
+}
+
+function storeCrashMessage(message) {
+    sessionStorage.setItem("crash-message", message)
+}
+
+function recoveredFromCrash() {
+    sessionStorage.removeItem("crash-message")
+}
+
+
+// === Crash detection ===
+
+function setupCrashDetection() {
+    Error.stackTraceLimit = Infinity
+
+    window.onerror = function (message, _url, _line, _column, error) {
```

Comment:
@s9ferech Could you elaborate more? First of all, I did not know that there is runtime difference between these two. Secondly, if this difference exists there and we have "slippery" situation of handlers that may not be run because of window refresh, this should be relaly well documented in the code, so no one will refactor it in the future in wrong way

Reply:
I just tested this and it turns out that my assumptions were wrong. This assignment seems to be equivalent to `addEventListener` indeed, except that it can be overwritten.

As far as I know, we only use one single listener on this event at the moment, and it seems very unlikely that we would add another one. For that reason, I think that it is no big problem anyway.